src/cettia.es6

```diff
@@ -0,0 +1,1259 @@
+      }
+    })
+    .on("open", function() {
+      // From connecting state
+      state = "opened";
+      var heartbeatTimer;
+      // Sets a heartbeat timer and clears it on close event
+      (function setHeartbeatTimer() {
+        // heartbeat event will be sent after options.heartbeat - options._heartbeat ms
+        heartbeatTimer = setTimeout(function() {
+          self.send("heartbeat").once("heartbeat", function() {
+            clearTimeout(heartbeatTimer);
+            setHeartbeatTimer();
+          });
+          // transport will be closed after options._heartbeat ms unless the server responds it
+          heartbeatTimer = setTimeout(function() {
+            self.fire("error", new Error("heartbeat"));
+            // Now that the transport doesn't realize its connection is closed, execute close method
+            // It will also fire close event to transport and accordingly socket
+            transport.close();
+          }, options._heartbeat);
+        }, options.heartbeat - options._heartbeat);
+      })();
+      self.once("close", function() {
+        clearTimeout(heartbeatTimer);
+      });
+      // Locks the connecting event
+      events.connecting.lock();
+      // Initializes variables related with reconnection
+      reconnectTimer = reconnectDelay = null;
+      reconnectTry = 0;
+    })
+    .on("close", function() {
+      // From connecting or opened state
+      state = "closed";
+      // Locks event whose order is lower than close event among reserved events
+      events.connecting.lock();
+      events.open.lock();
+
+      // Schedules reconnection
+      if (options.reconnect) {
+        // By adding a handler by one method in event handling
+        // it will be the last one of close event handlers having been added
+        self.once("close", function() {
+          reconnectDelay = options.reconnect.call(self, reconnectDelay, reconnectTry);
+          if (reconnectDelay !== false) {
+            reconnectTry++;
+            reconnectTimer = setTimeout(function() {
+              self.open();
+            }, reconnectDelay);
+            self.fire("waiting", reconnectDelay, reconnectTry);
+          }
+        });
+      }
+    })
+    .on("waiting", function() {
+      // From closed state
+      state = "waiting";
+    });
+
+  // Messaging
+  // A map for reply callback
+  var callbacks = {};
+  // Sends an event to the server via the connection
+  self.send = function(type, data, onResolved, onRejected) {
+    if (state !== "opened") {
+      self.fire("cache", [type, data, onResolved, onRejected]);
+      return this;
+    }
+
+    // Outbound event
+    var event = {
+      id: guid++,
+      type: type,
+      data: data,
+      reply: !!(onResolved || onRejected)
+    };
+    if (event.reply) {
+      callbacks[event.id] = [onResolved, onRejected];
+    }
+
+    // Determines if the given data contains binary
+    var hasBinary = false;
+    if (process.env.NODE_ENV !== "browser") {
```

Comment:
This used to be `typeof module.exports === 'object'`.  Now we can check node env variables directly.